libnmstate/state.py

```diff
@@ -155,6 +225,76 @@ def merge_interfaces(self, other_state):
             dict_update(other_state.interfaces[name], self.interfaces[name])
             self._ifaces_state[name] = other_state.interfaces[name]
 
+    def merge_route_config(self, other):
+        """
+        Merge routes from other to self.
+        If interface does exists in self, create an empty one.
+        Delete route entries if 'state: absent'.
+        If certain property missing when deleting, treat it as wildcard.
+        """
+        origin_other = copy.deepcopy(other)
+        # Copy other to self if certain interface does not defined.
+        for iface_name, routes in six.viewitems(other.iface_routes):
```

Comment:
@EdDev said there will be a lot of routes and it seems that there are a lot of loops over all routes. I guess this will need to be improved if there are going to be really a lot of routes. Maybe we should already add an integration test with many routes to see how this behaves.

Reply:
Jira issue created for performance bench-marking and improvement:

https://nmstate.atlassian.net/browse/NMSTATE-178